extensions/ros2/launch

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+function source_with_prefix() {
+    export COLCON_CURRENT_PREFIX="$1"
+    if [ -f "$COLCON_CURRENT_PREFIX/local_setup.bash" ]; then
```

Comment:
Is there a reason you're using `local_setup.bash` as opposed to `setup.sh` like the colcon plugin?

Reply:
This script should be the equivalent of setup.sh as I understand it?

Reply:
Hrm, we aren't we the ones chaining the overlay and underlay?  In this case, this launch script is supposed to effectively be that `setup.sh`...  I had reviewed the one generated by colcon:

```
# generated from colcon_bash/shell/template/prefix_chain.bash.em

# This script extends the environment with the environment of other prefix
# paths which were sourced when this file was generated as well as all packages
# contained in this prefix path.

# function to source another script with conditional trace output
# first argument: the path of the script
_colcon_prefix_chain_bash_source_script() {
  if [ -f "$1" ]; then
    if [ -n "$COLCON_TRACE" ]; then
      echo ". \"$1\""
    fi
    . "$1"
  else
    echo "not found: \"$1\"" 1>&2
  fi
}

# source chained prefixes
# setting COLCON_CURRENT_PREFIX avoids determining the prefix in the sourced script
COLCON_CURRENT_PREFIX="/opt/ros/foxy"
_colcon_prefix_chain_bash_source_script "$COLCON_CURRENT_PREFIX/local_setup.bash"

# source this prefix
# setting COLCON_CURRENT_PREFIX avoids determining the prefix in the sourced script
COLCON_CURRENT_PREFIX="$(builtin cd "`dirname "${BASH_SOURCE[0]}"`" > /dev/null && pwd)"
_colcon_prefix_chain_bash_source_script "$COLCON_CURRENT_PREFIX/local_setup.bash"

unset COLCON_CURRENT_PREFIX
unset _colcon_prefix_chain_bash_source_script
```

We could use that directly, but then would need patch it to handle `$SNAP`, which I thought was more painful than just handling it in the launcher?  If I'm missing something, we can certainly use the setup.sh, but I didn't see a good way to prescribe the $SNAP without some ugly sed foo.

Reply:
Okay, let's just keep an eye on this then. Especially if we add the ability to find an underlay in the staging area like the catkin plugin does.